### PR TITLE
[Binning e2e coverage] Longitude Correctness

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Adds a series of tests around Longitude correctness, meaning that it checks rendered values for each of the bucket sizes.

1. Makes sure correct visualization renders
2. Makes sure correct axis labels render
3. Makes sure correct x-axis "ticks" (individual value labels) render
4. Makes sure the selected bucket size is correctly displayed in the sidebar

### Screenshots
image

### CODE REVIEW

File: frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    cy.get(".bar");
+    cy.get(".y-axis-label")
+      .invoke("text")
+      .should("eq", "Count");
+    cy.get(".x-axis-label")
+      .invoke("text")
+      .should("eq", "Longitude");
+
+    cy.get(".axis.x").within(() => {
+      cy.findByText("170° W");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js:26]: @flamber @ariya - I was thinking to literary pass the array of all X axis "ticks" for each of the binning buckets to make sure they render correctly.
Do you think it's an overkill?

Alternatively, I can pick just a few representative values and make sure they render.

File: frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      cy.findByText("Auto bin").click();
+    });
+    getTitle("Count by Longitude: Auto binned");
+    cy.get(".bar");
+    cy.get(".y-axis-label")
+      .invoke("text")
+      .should("eq", "Count");
+    cy.get(".x-axis-label")
+      .invoke("text")
+      .should("eq", "Longitude");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js:23]: @flamber Is it ok if we assert on the labels just in one test or would you like to have this check in each of the tests here?

File: frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js (hunk 1 above)
Comment (flamber) [reply frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js:26]: I would just pick the `100°` tick and then get the tick before and after. Or the first three ticks, but at 1° might skip some degrees

File: frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js (hunk 2 above)
Comment (flamber) [reply frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js:23]: One time should be plenty. These should not change based on granularity.

File: frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const LONGITUDE_OPTIONS = {
   },
   "Bin every 1 degree": {
     selected: "1°",
-    representativeValues: ["167° W", "166° W", "60° W"],
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js]: Had to update this, but it feels a bit weird. I would've expected there to be a "166° W" value in the chart based on these screenshots.

image

image

File: frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js (hunk 1 above)
Comment (ariya) [reply frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js]: I agree with @flamber, just sample some ticks.

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/binning/correctness/longitude.cy.spec.js]: This was resolved. I'll mark the discussion as such.
Thanks

**Review by ariya — APPROVED** [review]

**Pull request merged**